src/DurableTask.Core/TaskOrchestrationExecutor.cs

```
@@ -87,9 +89,16 @@ IEnumerable<OrchestratorAction> ExecuteCore(IEnumerable<HistoryEvent> eventHisto
                         {
                             if (this.result.IsFaulted)
                             {
-                                Debug.Assert(this.result.Exception != null);
-
-                                this.context.FailOrchestration(this.result.Exception.InnerExceptions.FirstOrDefault());
+                                Exception exception = this.result.Exception?.InnerExceptions.FirstOrDefault();
+                                Debug.Assert(exception != null);
+
+                                if (Utils.IsExecutionAborting(exception))
+                                {
+                                    // Let this exception propogate out to be handled by the dispatcher
+                                    this.result.GetAwaiter().GetResult();
```

It's not exactly clear to me that this will let the exception propagate like the comment states. Is there a reason this is preferred to something like the below?

`ExceptionDispatchInfo.Capture(exception).Throw();`

---

It was a toss-up to me. I can change it if that makes it more clear.

---

I think I prefer the explicit throw, as then I don't need to read the comment to realize that this line will throw an exception.